jobs/integration/validation.py

```diff
@@ -111,10 +112,10 @@ async def run_until_success(unit, cmd, timeout_insec=None):
             click.echo(
                 "Action " + action.status + ". Command failed on unit " + unit.entity_id
             )
-            click.echo("cmd: " + cmd)
-            click.echo("code: " + action.code)
-            click.echo("stdout:\n" + action.stdout.strip())
-            click.echo("stderr:\n" + action.stderr.strip())
+            click.echo(f"cmd: {cmd}")
+            click.echo(f"code: {action.code}")
+            click.echo(f"stdout:\n{action.stdout}")
+            click.echo(f"stderr:\n{action.stderr}")
```

Comment:
prior juju-3.x changes broke this when action.code was an int

str + int is invalid -- swapped to an f-string